content/en/docs/Architecture/cloud-init.md

```diff
@@ -41,32 +41,51 @@ stages:
+| _before-install_           | This stage happens after partitioning but before the image OS is applied                                                                                                                                                                                                            |
+| _after-install-chroot_     | This stage happens after installing active and grub                                                                                                                                                                                                                                 |
+| _after-install_            | This stage runs after active,passive and recovery images are installed and after disks have been encrypted                                                                                                                                                                          |
+| _before-reset_             | This stage happens after partitions have been formatted and mounted but before the image has been resetted                                                                                                                                                                          |
+| _after-reset_              | This stage happens after partitions have been formatted and mounted and active and passive images resetted                                                                                                                                                                          |
+| _after-reset-chroot_       | This stage happens after active has been reset but before passive has been touched.                                                                                                                                                                                                 |
+| _before-upgrade_           | This stage happens after mounting partitions with RW but before any image has been upgraded                                                                                                                                                                                         |
+| _after-upgrade_            | This stage happens after upgrade has been done                                                                                                                                                                                                                                      |
+| _after-upgrade-chroot_     | This stage happens after the image has been upgraded                                                                                                                                                                                                                                |
```

Comment:
what's the difference with the previous one? Maybe explain here?

Reply:
its explained at the end that all chroot stages happens in a chroot inside the affected image :D

Reply:
@jimmykarily if you do the diff "without white space" it's a bit clearer to see what was changed, and what was added

Reply:
@mauromorales by "previous" one I meant the one above not the previous "version". Sorry for the confusion. It's a bit explained below but still the user will read these 2 in sequence and will only have to trust that we explain later and keep reading. Ideally the text shouldn't read so similar between these two.

Reply:
Maybe in the chroot ones we can have a link to the footnote or something so we can jump directly to the comment?

Reply:
That would work too. A here-link with text "How is this different from `after-upgrade`?"